hazelcast/src/main/java/com/hazelcast/jet/impl/MasterJobContext.java

```diff
@@ -697,6 +697,10 @@ private Throwable getErrorFromResponses(String opName, Collection<Map.Entry<Memb
 
         if (!otherFailures.isEmpty()) {
             return (Throwable) otherFailures.get(0).getValue();
+        } else if (!isJobSuspendable(mc.jobConfig())) {
+            return topologyFailures.get(0).getValue() instanceof CancellationException
```

Comment:
job can be cancelled for many reasons, eg. member terminating. I am not sure if mapping CancellationException to CancellationByUserException is correct.

Reply:
Exactly. I wrote in the description

> Worth noticing, that, in theory, there also might be other specific job cancellation causes with CancellationException, so that patch mostly is 'in discussion' state.

Reply:
did you check what exceptions are used in different situations?

Reply:
That's what am I actually doing. 
@k-jamroz  Just curious - can we live with existing inconvenience? I have a gut feeling the fix won't be such trivial.

Reply:
https://github.com/hazelcast/hazelcast/pull/18498/files

Caught. Thinking how to deal with it.

Reply:
https://github.com/hazelcast/hazelcast/pull/26037/commits/df179d04086577aa7948c78267060f61a7b157ab

Boom ;)